Release build failure

## Expected Behavior

Building a Release configuration of HPX should work

## Actual Behavior

Only Debug builds are successful.  Both Release and RelWithDebInfo builds fail.  See http://omega.nic.uoregon.edu:8020/#/builders/6/builds/163/steps/7/logs/stdio

```
[ 91%] Linking CXX shared library ../lib/libhpx.so
../lib/libhpx_serialization.a(exception_ptr.cpp.o): In function `hpx::serialization::detail::load(hpx::serialization::input_archive&, std::__exception_ptr::exception_ptr&, unsigned int)':
exception_ptr.cpp:(.text+0xc92): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<std::runtime_error>(std::runtime_error const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0xd52): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<std::invalid_argument>(std::invalid_argument const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0xe0a): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<std::out_of_range>(std::out_of_range const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0xec2): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<std::logic_error>(std::logic_error const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0x122b): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<boost::system::system_error>(boost::system::system_error const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0x1511): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<hpx::detail::std_exception>(hpx::detail::std_exception const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0x15f1): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<hpx::detail::bad_exception>(hpx::detail::bad_exception const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0x16d1): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<hpx::detail::bad_alloc>(hpx::detail::bad_alloc const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0x17b1): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<hpx::detail::bad_typeid>(hpx::detail::bad_typeid const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
exception_ptr.cpp:(.text+0x1891): undefined reference to `std::__exception_ptr::exception_ptr hpx::detail::get_exception<hpx::detail::bad_cast>(hpx::detail::bad_cast const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [lib/libhpx.so.1.5.0] Error 1
make[1]: *** [src/CMakeFiles/hpx_core.dir/all] Error 2
make: *** [all] Error 2
```

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1.  Configure HPX with Release or RelWithDebInfo
  1. Build

## Specifications

... Please describe your environment

  - HPX Version:  current master
  - Platform (compiler, OS):  GCC 7.3.0 on RHEL

--- a/libs/errors/src/exception.cpp
+++ b/libs/errors/src/exception.cpp
@@ -276,6 +276,50 @@ namespace hpx { namespace detail {
     template HPX_EXPORT std::exception_ptr get_exception(hpx::exception const&,
         std::string const&, std::string const&, long, std::string const&);
 
+    template HPX_EXPORT std::exception_ptr get_exception(
+        boost::system::system_error const&, std::string const&,
+        std::string const&, long, std::string const&);
+
+    template HPX_EXPORT std::exception_ptr get_exception(std::exception const&,
+        std::string const&, std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::std_exception const&, std::string const&,
+        std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::bad_exception const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_exception const&, std::string const&,
+        std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(std::bad_typeid const&,
+        std::string const&, std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_typeid const&, std::string const&, std::string const&,
+        long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(std::bad_cast const&,
+        std::string const&, std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_cast const&, std::string const&, std::string const&,
+        long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(std::bad_alloc const&,
+        std::string const&, std::string const&, long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        hpx::detail::bad_alloc const&, std::string const&, std::string const&,
+        long, std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::logic_error const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::runtime_error const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::out_of_range const&, std::string const&, std::string const&, long,
+        std::string const&);
+    template HPX_EXPORT std::exception_ptr get_exception(
+        std::invalid_argument const&, std::string const&, std::string const&,
+        long, std::string const&);
+
+    ///////////////////////////////////////////////////////////////////////////
     template HPX_EXPORT void throw_exception(
         hpx::exception const&, std::string const&, std::string const&, long);
 

